sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,423 @@
+    TEASER(...)
+    >>> y_pred = clf.predict(X_test)
+    """
+
+    _tags = {
+        "capability:multivariate": True,
+        "capability:multithreading": True,
+        "capability:early_prediction": True,
+    }
+
+    def __init__(
+        self,
+        estimator=None,
+        classification_points=None,
+        n_jobs=1,
+        random_state=None,
+    ):
+        self.estimator = estimator
+        self.classification_points = classification_points
+
+        self.n_jobs = n_jobs
+        self.random_state = random_state
+
+        self._estimators = []
+        self._classification_points = []
+        self._svms = []
+        self._consecutive_predictions = 0
+
+        self._svm_gammas = [100, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1.5, 1]
+
+        super(TEASER, self).__init__()
+
+    def _fit(self, X, y):
+        n_instances, _, series_length = X.shape
+
+        self._classification_points = (
+            copy.deepcopy(self.classification_points)
+            if self.classification_points is not None
+            else [round(series_length / i) for i in range(1, 21)]
+        )
+        # remove duplicates
+        self._classification_points = list(set(self._classification_points))
+        self._classification_points.sort()
+        # remove classification points that are less than 3
+        self._classification_points = [i for i in self._classification_points if i >= 3]
+        # make sure the full series length is included
+        if self._classification_points[-1] != series_length:
+            self._classification_points.append(series_length)
+        # create dictionary of classification point indicies
+        self._classification_point_dictionary = {}
+        for index, classification_point in enumerate(self._classification_points):
+            self._classification_point_dictionary[classification_point] = index
+
+        m = getattr(self.estimator, "n_jobs", None)
+        threads = self._threads_to_use if m is None else 1
+
+        fit = Parallel(n_jobs=threads)(
+            delayed(self._fit_estimator)(
+                X,
+                y,
+                i,
+            )
+            for i in range(len(self._classification_points))
+        )
+
+        self._estimators, self._svms, X_svm, train_preds = zip(*fit)
+
+        # tune consecutive predictions required to best harmonic mean
+        best_hm = -1
+        for g in range(2, min(6, len(self._classification_points))):
+            state_info = [(0, 0, 0) for _ in range(n_instances)]
+            finished = [False for _ in range(n_instances)]
+
+            for i in range(len(self._classification_points)):
+                if i == len(self._classification_points) - 1:
+                    decisions = [True for _ in range(n_instances)]
+                elif self._svms[i] is not None:
+                    decisions = self._svms[i].predict(X_svm[i]) == 1
+                else:
+                    decisions = [False for _ in range(n_instances)]
```

Comment:
Maybe initialize decisions to False in line 153?
 (instead of having one branch in the if  ... else ...?

Reply:
Don't think it's necessary to create the list twice, setting everything else should not happen often. But it's a minor optimisation either way.

Reply:
Ok, but what about this simplification 

```
if self._svms[i] is not None:
    decisions = self._svms[i].predict(X_svm[i]) == 1
else:
    decisions = [i == len(self._classification_points) - 1 for _ in range(n_instances)]
```

Reply:
This does not create twice but is still more compact

Reply:
Thinking about my suggestion, maybe it is not the same as your code ...

Reply:
Agreed